### PR TITLE
Update Explainer with latest changes to the API

### PR DESCRIPTION
**Pull request by majido**
Wrote small sections on: WorkletAnimations, ScrollTimline, multiple timeline, group effect, and migration mechanism.

Added examples following latest syntax changes 1) Twitter header which is a scroll-linked effect with multiple elements 2) hidey-bar which is an effect that uses multiple timelines but a single element. 

WIP.md now contains only remaining open questions

### CODE REVIEW

**Comment by majido** [comment]
@flackr  @stephenmcgruer FYI ^^^

File: README.md [hunk 1]
```diff
@@ -55,14 +57,263 @@ web platform features enabled (`--enable-experimental-web-platform-features`
 flag) otherwise they fallback to using main thread rAF to emulate the behaviour.
 
 
+# Key Concepts
+
+## WrokletAnimation
```
Comment (stephenmcgruer) [inline README.md]: Spelling error

File: README.md [hunk 2]
```diff
@@ -55,14 +57,263 @@ web platform features enabled (`--enable-experimental-web-platform-features`
...
+`WorkletAnimation` is a subclass of Animation that can be used to create an custom animation effect
+that runs inside the animation worklet scope. A worklet animation has a  corresponding animator
```
Comment (stephenmcgruer) [inline README.md]: 'the animation worklet scope' isn't defined yet, perhaps something like 'runs inside a {separate,standalone?} animation worklet scope'?

Nit; double-space before 'corresponding'.

File: README.md [hunk 3]
```diff
@@ -55,14 +57,263 @@ web platform features enabled (`--enable-experimental-web-platform-features`
...
+instance in the animation worklet scope which is responsible to drive its keyframe effects.
+Here are the key differences compared to a regular web animation:
+  - AnimationId should match a specific animatior class registered in worklet scope.
```
Comment (stephenmcgruer) [inline README.md]: Nit; 'the animation worklet scope'

File: README.md [hunk 4]
```diff
@@ -55,14 +57,263 @@ web platform features enabled (`--enable-experimental-web-platform-features`
...
+  - WorkletAnimation may have multiple timelines (including ScrollTimelines).
+  - WorkletAnimation may have a custom properties bag that can be cloned and provided to animator
+    constructor when it is being instantiated.
+
+Note that worklet animations expose same API aurface as other web animations and thus they may be
```
Comment (stephenmcgruer) [inline README.md]: s/aurface/surface

File: README.md [hunk 5]
```diff
@@ -55,14 +57,263 @@ web platform features enabled (`--enable-experimental-web-platform-features`
...
+worklet animations and it is the intended mechanisms to give read access to scroll position.
+
+## GroupEffect
+[GroupEffect](https://w3c.github.io/web-animations/level-2/#the-animationgroup-interfaces) is a
+concept introduced in Web Animation Level 2 specification. It provides a way to group multiple
+effects in a tree structure. GroupEffect can be used as the output for worklet animations. It
+makes it possible for worklet animation to drive effects spanning multiple elements.
+
+**TODO**: At the moment, group effect define strict rules to translate group effect time to children
+effect times but it given flexibility of children start time. This is not flexible enough for
```
Comment (stephenmcgruer) [inline README.md]: Not sure what 'it given flexibility' means here.

File: README.md [hunk 6]
```diff
@@ -55,14 +57,263 @@ web platform features enabled (`--enable-experimental-web-platform-features`
...
+new WorklerAnimation('animation-with-local-state', [], [], {value: 1});
+```
+
+
+```js
+registerAnimator('animation-with-local-state', class {
+  constructor(options) {
+    // options is:
+    //  1. the user provided options bag to WorkletAnimation constructor on first initialization i.e, {value: 1}.
+    //  2. the object returned by |destory| after each migration i.e. {value: 42}.
```
Comment (stephenmcgruer) [inline README.md]: s/destory/destroy

File: README.md [hunk 7]
```diff
@@ -55,14 +57,263 @@ web platform features enabled (`--enable-experimental-web-platform-features`
...
+```js
+// in document scope
+new WorklerAnimation('animation-with-local-state', [], [], {value: 1});
+```
+
+
+```js
+registerAnimator('animation-with-local-state', class {
+  constructor(options) {
+    // options is:
```
Comment (stephenmcgruer) [inline README.md]: s/is/may be

File: README.md (hunk 1 above)
Comment (majido) [reply README.md]: done.

File: README.md (hunk 2 above)
Comment (majido) [reply README.md]: Added a section to define Animation Worklet Scope

File: README.md (hunk 3 above)
Comment (majido) [reply README.md]: Done.

File: README.md (hunk 4 above)
Comment (majido) [reply README.md]: Done.

File: README.md (hunk 5 above)
Comment (majido) [reply README.md]: Explained in more details what current limitation of GroupEffect is and what the flexibility we want to add translates to.

File: README.md (hunk 7 above)
Comment (majido) [reply README.md]: Done.

File: README.md (hunk 6 above)
Comment (majido) [reply README.md]: Done.

**Review by stephenmcgruer — APPROVED** [review]

**Pull request merged**